### PR TITLE
[DERCBOT-538] :sparkles: add a path for a summary story search

### PR DESCRIPTION
**Pull request by djowalker**

### CODE REVIEW

File: bot/admin/server/src/main/kotlin/model/SummaryStorySearchRequest.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import ai.tock.bot.admin.story.StoryDefinitionConfigurationMinimalSummaryRequest
+import ai.tock.nlp.admin.model.ApplicationScopedQuery
+
+/**
+ * Minimal story search request without pagination
+ * Used principally into custom metrics search component
+ */
+class SummaryStorySearchRequest(
+    val category: String? = null,
```
Comment (Benvii) [inline bot/admin/server/src/main/kotlin/model/SummaryStorySearchRequest.kt:27]: In the ticket comment I also notice a `textSearch` field here : https://jiradc.intra.arkea.com:8443/jira/browse/DERCBOT-538?focusedCommentId=6337848&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-6337848

I don't know if it's required or not, I don't think that we need it.

File: bot/admin/server/src/main/kotlin/BotAdminVerticle.kt [hunk 2]
```diff
@@ -610,6 +611,28 @@ open class BotAdminVerticle : AdminVerticle() {
...
+                if (request.applicationName.isEmpty()) {
+                    badRequest("applicationName is needed")
+                }
+                BotAdminService.searchSummaryStories(request)
+            } else {
+                unauthorized()
+            }
+        }
+
+        blockingJsonPost("/bot/story/search", setOf(botUser, faqBotUser)) { context, request: StorySearchRequest ->
```
Comment (Benvii) [inline bot/admin/server/src/main/kotlin/BotAdminVerticle.kt:628]: Why is it duplicated here ? it's already present at line 606-605 ?

**Review by Benvii — CHANGES_REQUESTED** [review]
Ok for me small code duplicate and a question about a field mentioned on the ticket not implemented here but that doesn't seems useful.
There is no tests, I noticed that BotAdminService.search* isn't currently tested.

File: bot/admin/server/src/main/kotlin/model/SummaryStorySearchRequest.kt (hunk 1 above)
Comment (djowalker) [reply bot/admin/server/src/main/kotlin/model/SummaryStorySearchRequest.kt:27]: `textSearch` is not required the first comments were before a discussion with @rkuffer

File: bot/admin/server/src/main/kotlin/BotAdminVerticle.kt (hunk 2 above)
Comment (djowalker) [reply bot/admin/server/src/main/kotlin/BotAdminVerticle.kt:628]: a wrong copye/paste i think, i erased it.

**Comment by djowalker** [comment]
Added some mocked tests about search summary however there are not really interesting

**Review by Benvii — APPROVED** [review]
LGTM, thanks for the tests and the update 👍️

**Pull request merged**